main/src/ca/uwaterloo/flix/language/phase/Reducer.scala

```diff
@@ -42,10 +42,12 @@ object Reducer {
 
     val newDefs = ParOps.parMapValues(root.defs)(visitDef)
     val defTypes = ctx.defTypes.keys.asScala.toSet
-
+    // This is an over approximation of the types in enums and structs since they are erased.
+    val enumTypes = Set(MonoType.Bool, MonoType.Char, MonoType.Int8, MonoType.Int16, MonoType.Int32, MonoType.Int64, MonoType.Float32, MonoType.Float64, MonoType.Object)
+    val structTypes = enumTypes
     val effectTypes = root.effects.values.toSet.flatMap(typesOfEffect)
 
-    val types = nestedTypesOf(Set.empty, Queue.from(defTypes ++ effectTypes))
+    val types = nestedTypesOf(Set.empty, Queue.from(effectTypes ++ structTypes ++ enumTypes ++ defTypes))
```

Comment:
I dream of the day we can get rid of nestedTypesOf :O

Reply:
It's not pretty but it seems required in some form to figure out what tuples and functions to generate